lib/SILOptimizer/Transforms/RedundantLoadElimination.cpp

```diff
@@ -1089,6 +1092,18 @@ void BlockState::processInstructionWithKind(RLEContext &Ctx,
     return;
   }
 
+  // Begin access writes to memory only if one of its users writes to memory.
+  // We will look at all of its users and we can project to the source memory
+  // location so, if there are any actual writes to memory we will catch them
+  // there so, we can ignore begin_access here.
+  if (isa<BeginAccessInst>(Inst))
+    return;
+
+  // If the load is valid, then this strong release won't invoke the destructor.
+  // So we can ignore it.
```

Comment:
I don't get this argument.

Reply:
I didn't give much context, sorry. I'll try to explain my reasoning a bit more clearly. The only case where we'd care about the strong_release is when it turns into a destructor with a read/write in it (in this case, only a write, actually). If this load is valid (this load being whatever load we happen to be promoting at any given time), then the strong_release can't have destroyed the objected we're loading before we load it. Does that make sense?